### PR TITLE
make BinaryLabels can save the values parameter

### PR DESCRIPTION
**Pull request by MikeLing**

### CODE REVIEW

**Comment by vigsterkr** [comment]
@MikeLing adding a test for this (serialization test) would be good so that we know it works :P

**Comment by MikeLing** [comment]
mmm, I just create a serialization test for this. But, this https://gist.github.com/MikeLing/384a50fa1b2267e9a14b5fcf67e59c5d passed on develop branch(without this patch). So I'm wondering how to prove this works except to test it on notebook

**Comment by vigsterkr** [comment]
can you paste here the content of the `serialization-asciiCBinaryLabels.XXXXXX` file?

**Comment by MikeLing** [comment]
sure!

<<_SHOGUN_SERIALIZABLE_ASCII_FILE_V_00_>>
subset_stack SGSerializable* SubsetStack [
active_subset SGSerializable* null []
active_subsets_stack SGSerializable* DynamicObjectArray [
array Vector<SGSerializable*> 0 ()
resize_granularity int32 128
use_sg_malloc bool t
free_array bool t
dim1_size int32 1
dim2_size int32 1
dim3_size int32 1
]
]
labels SGVector<float64> 10 ({6.923869082654833e-310}{6.923869082654833e-310}{1.581010066691989e-322}{4.743030200075967e-322}{4.685135602302451e-310}{6.923869082642976e-310}{3.162020133383978e-322}{3.162020133383978e-322}{4.685135602302451e-310}{4.685135602265297e-310})

**Comment by vigsterkr** [comment]
@MikeLing you are testing the wrong thing in you test... you should check `get_value(i)` equaling...
```
	for (int32_t i = 0; i < new_labels->get_num_labels(); i++)
	{
		EXPECT_EQ(
		    labels->get_value(i), new_labels->get_value(i));
}
```

File: tests/unit/labels/BinaryLabels_unittest.cc [hunk 1]
```diff
@@ -32,3 +34,35 @@ TEST(BinaryLabels,scores_to_probabilities)
 
 	SG_UNREF(labels);
 }
+
+TEST(BinaryLabels, serialization)
+{
+	CBinaryLabels* labels=new CBinaryLabels(10);
+	labels->set_values(SGVector<float64_t>(labels->get_num_labels()));
```
Comment (vigsterkr) [inline tests/unit/labels/BinaryLabels_unittest.cc]: what's this?!

File: tests/unit/labels/BinaryLabels_unittest.cc [hunk 2]
```diff
@@ -32,3 +34,35 @@ TEST(BinaryLabels,scores_to_probabilities)
...
+
+	for (index_t i=0; i<labels->get_num_labels(); ++i)
```
Comment (vigsterkr) [inline tests/unit/labels/BinaryLabels_unittest.cc]: could you instead just make sure that you create an `v = SGVector<float64_t>(labels->get_num_labels())` and set it random... and basically just do:
```
labels->set_values(v);
labels->set_labels(v);
```

File: tests/unit/labels/BinaryLabels_unittest.cc [hunk 3]
```diff
@@ -32,3 +34,35 @@ TEST(BinaryLabels,scores_to_probabilities)
...
+	CBinaryLabels* new_labels = new CBinaryLabels;
+	new_labels->load_serializable(file);
+	file->close();
+	SG_UNREF(file);
+
+	ASSERT(new_labels->get_num_labels () == 10)
+
+	for (int32_t i = 0; i < new_labels->get_num_labels(); i++)
+	{
+		EXPECT_EQ(
```
Comment (vigsterkr) [inline tests/unit/labels/BinaryLabels_unittest.cc]: here check both that values and labels are equal.

**Pull request merged**